kong/plugins/apifortress/fortress_serializer.lua

```diff
@@ -0,0 +1,117 @@
+-- The serializer. Builds a data structure that is compatible with
+-- the API Fortress API expectations
+
+local stringy = require "stringy"
+
+local _M = {}
+
+-- pre-indexing global functions
+local str_startswith = stringy.startswith
+local str_endswith = stringy.endswith
+local str_sub = string.sub
+local str_len = string.len
+local table_insert = table.insert
+local encode_base64 = ngx.encode_base64
+
+function _M.serialize(ngx)
+
+  -- pre-indexing to reduce the number of lookups later on
+  local ngx_req = ngx.req
+  local ngx_resp = ngx.resp
+  local ngx_var = ngx.var
+  local ngx_ctx = ngx.ctx
```

Comment:
In the openresty documentation I find this example:

    -- mymodule.lua
    local _M = {}
   
    -- the following line is bad since ngx.ctx is a per-request
    -- data while this `ctx` variable is on the Lua module level
    -- and thus is per-nginx-worker.
    local ctx = ngx.ctx
   
    function _M.main()
        ctx.foo = "bar"
    end
   
    return _M

And this is plain logic to me. You can't store the context in a variable as it would end up shared. I understand this as a bad behavior. But I don't think I did it.
I will simply fix the code if you ask me to, but I'd like to understand.